components/infobox/wikis/warcraft/infobox_patch_custom.lua

```diff
@@ -0,0 +1,142 @@
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Cell = Widgets.Cell
+
+local CustomPatch = Class.new()
+
+local BALANCE_UPDATE = 'Balance Update'
+local IGNORE_NET_EASE_RELEASE_BEFORE = DateExt.readTimestamp('2011-03-23')
+local SKIP = 'skip'
+local SKIPPED = 'skipped'
+
+local _args
+
+local CustomInjector = Class.new(Injector)
+
+function CustomPatch.run(frame)
+	local customPatch = Patch(frame)
+	_args = customPatch.args
+
+	customPatch.createWidgetInjector = CustomPatch.createWidgetInjector
+	customPatch.getChronologyData = CustomPatch.getChronologyData
+	customPatch.addToLpdb = CustomPatch.addToLpdb
+
+	return customPatch:createInfobox()
+end
+
+function CustomPatch:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'release' then
+		return {
+			Cell{name = '[[Public Test Realm|PTR]] Release Date', content = {_args.release_ptr}},
+			Cell{name = 'Release Date', content = {_args.release}},
+			Cell{name = '[[NetEase]] Release Date', content = {CustomPatch._netEaseRelease(_args)}},
+		}
+	end
+
+	return widgets
+end
+
+function CustomPatch._netEaseRelease(args)
+	if not args.release or DateExt.readTimestamp(args.release) <= IGNORE_NET_EASE_RELEASE_BEFORE then
+		return
+	end
+
+	return args.release_netease == SKIP and SKIPPED or args.release_netease
+end
+
+function CustomPatch:addToLpdb(args)
+	mw.ext.LiquipediaDB.lpdb_datapoint('patch_' .. self.pagename, {
+		name = self.name,
+		type = 'patch',
+		date = args.release,
+		extradata = mw.ext.LiquipediaDB.lpdb_create_json({
+			beta = tostring(Logic.readBool(args.beta)),
+			version = self.name,
+			release = args.release,
+			neteaserelease = args.release_netease ~= SKIP and args.release_netease or nil,
+			ptrdate = args.release_ptr,
+			balanceupdates = tostring(CustomPatch._hasBalanceUpdate(args)),
+			previous = args.previous and ('Patch args.previous') or nil,
+			next = args.next and ('Patch args.previous') or nil,
+		})
+	})
+end
+
+function CustomPatch:getChronologyData(args)
+	local previousBalanceUpdate, nextBalanceUpdate = CustomPatch._automaticChronologyData(args)
+
+	local toBalanceUpdateLink = function(input)
+		return input and (input .. '#Balance_Updates|' .. input) or nil
+	end
+
+	local toPatch = function(input)
+		return input and ('Patch ' .. input .. '|' .. input) or nil
+	end
+
+	if not args.previous and not args.next then
+		return {
+			previous = toBalanceUpdateLink(previousBalanceUpdate),
+			next = toBalanceUpdateLink(nextBalanceUpdate),
+		}
+	end
+
+	return {
+		previous = toPatch(args.previous),
+		next = toPatch(args.next),
+		previous2 = toBalanceUpdateLink(previousBalanceUpdate),
+		next2 = toBalanceUpdateLink(nextBalanceUpdate),
+	}
+end
+
+function CustomPatch._automaticChronologyData(args)
+	if not args.release or not CustomPatch._hasBalanceUpdate(args) then
+		return
+	end
+
+	local baseConditions = '[[type:patch]] AND [[extradata_balanceupdates::true]] AND [[date::<'
```

Comment:
```suggestion
	local baseConditions = '[[type:patch]] AND [[extradata_balanceupdates::true]] AND [[date::'
```
But not sure of putting half of the date statement here and other has as string concat a in the query itself.